### PR TITLE
Deprecate `length_multiplier`

### LINKED ISSUE #1349: deprecate `--length-multiplier` CLI option
`--length-multiplier` doesn't compose well with other options, and specifying a fixed length is an easier way to get short-running tests.

### PR DESCRIPTION
**Pull request by nspope**
Fixes #1349

### CODE REVIEW

**Comment by nspope** [comment]
Linting error in the docs,
```
 diff --git a/docs/tutorial.rst b/docs/tutorial.rst
index 36a3cce..bc5d619 100644
--- a/docs/tutorial.rst
+++ b/docs/tutorial.rst
@@ -512,8 +512,9 @@ This provides both a ``size_history`` plot and a ``tubes`` visualization. E.g.
 .. code-block:: python
 
     import demesdraw
+
     graph = model.model.to_demes()
-    demesdraw.tubes(model.model.to_demes());
+    demesdraw.tubes(model.model.to_demes())
 
 Equivalent plots for all the available demographic models are also
 shown in the :ref:`sec_catalog`.
```

though I think the semicolon is intended here?

**Comment by nspope** [comment]
I'd like to modify the tests so that they don't use `length_multiplier`, but forgot that it differs from `left` / `right` in that the latter keeps the original contig's coordinate system. This matters esp for SLiM, where the "masked" portions of a contig are actually simulated with recombination rate 0 (and can have impactful mutations added if there's a DFE).

File: CHANGELOG.rst [hunk 1]
```diff
@@ -1,3 +1,13 @@
+--------------------
+[0.2.1] - 2024-XX-XX
+--------------------
+
+**Breaking changes**:
+
+- The `length_multiplier` option to `Species.get_contig` is deprecated and
+  prints a warning. The options `left` and `right` should be used to truncate a
+  contig, instead. (:user:`nspope`, :pr:`xxxx`)
```
Comment (petrelharp) [inline CHANGELOG.rst:9]:
```suggestion
  contig, instead. (:user:`nspope`, :pr:`1605`)
```

File: stdpopsim/cli.py [hunk 2]
```diff
@@ -570,13 +570,16 @@ def add_simulate_species_parser(parser, species):
...
         "-l",
         "--length-multiplier",
-        default=1,
+        default=None,
         type=float,
-        help="Simulate a sequence of length l times the named chromosome's length "
-        "using the named chromosome's mutation and recombination rates.",
+        help="Deprecated. Simulate a sequence of length l times the named "
+        "chromosome's length using the named chromosome's mutation and "
+        "recombination rates.",
```
Comment (petrelharp) [inline stdpopsim/cli.py:582]:
```suggestion
        help="Deprecated: use left/right instead. Simulate a sequence of "
        "length l times the named chromosome's length using the named "
        "chromosome's mutation and recombination rates.",
```

File: stdpopsim/species.py [hunk 3]
```diff
@@ -207,11 +207,12 @@ def get_contig(
...
-            `length_multiplier` times the length of the specified chromosome with the
-            same chromosome-specific mutation and recombination rates.
-            This option cannot currently be used in conjunction with the
-            ``genetic_map`` argument.
+        :param float length_multiplier: Deprecated, use `left` and `right`
+            instead. If specified, simulate a region of length `length_multiplier`
+            times the length of the specified chromosome with the same
+            chromosome-specific mutation and recombination rates.  This option
+            cannot currently be used in conjunction with the ``genetic_map``
+            argument.
```
Comment (petrelharp) [inline stdpopsim/species.py:215]:
```suggestion
            cannot be used in conjunction with the ``genetic_map`` argument.
```

**Review by petrelharp — CHANGES_REQUESTED** [review]
LGTM!

**Comment by nspope** [comment]
Are we distinguishing 0.2.1 from 0.2.1a in the changelog? Looks like not, as the discretization bug fix went under 0.2.1a?

**Pull request merged**